Add more support for tags

clns opened the pull request:
Add `TagsCollection`.

Add support for `git_tag_list`, `git_tag_list_match`, `git_tag_foreach`, `git_tag_create_lightweight`.

carlosmn commented:
This moves the code back to having a lot of methods on `Repository`, which is a style which current git2go has abandoned. These should be methods on a `TagsCollection` object.

carlosmn commented on tag.go:
```diff
@@ -42,3 +48,129 @@ func (t Tag) TargetId() *Oid {
+// The created tag is a simple reference and can be queried using
+// repo.References.Lookup("refs/tags/<name>"). The name of the tag (eg "v1.0.0")
+// is queried with ref.Shorthand().
+func (repo *Repository) CreateTagLightweight(name string, commit *Commit, force bool) (*Oid, error) {
```
This needs to explain what the return Oid is, there is no way to know what it is from the signature.

clns replied:
I added more explanation in the next commit.

clns commented:
I noticed the other collections but since there wasn't any for tags I didn't know if I should add one. I can go ahead and add a `TagsCollection`, let me know.

carlosmn commented:
Yes, do add `TagsCollection`.

clns commented:
Added `TagsCollection` and moved the new methods on it.

This PR doesn't break any existing apis, so moving the existing tags-related methods from `Repository` into `TagsCollection` is outside of the scope of this PR and is a todo.
